conans/client/graph/graph_binaries.py

```diff
@@ -242,6 +221,33 @@ def _process_node(self, node, build_mode):
             if node.conanfile.info.invalid and node.conanfile.info.invalid[0] == BINARY_INVALID:
                 node.binary = BINARY_INVALID
 
+    def _process_compatible_node(self, node):
+        """ simplified checking of compatible_packages, that should be found existing, but
+        will never be built, for example. They cannot be editable either at this point.
+        """
+        # Check that this same reference hasn't already been checked
+        if self._evaluate_is_cached(node):
+            return
+
+        # TODO: Test that this works
+        if node.conanfile.info.invalid and node.conanfile.info.invalid[0] == BINARY_ERROR:
+            node.binary = BINARY_ERROR
+            return
+
+        # Obtain the cache_latest valid one, cleaning things if dirty
+        while True:
+            cache_latest_prev = self._cache.get_latest_package_reference(node.pref)
+            if cache_latest_prev is None:
+                break
+            package_layout = self._cache.pkg_layout(cache_latest_prev)
+            if not self._evaluate_clean_pkg_folder_dirty(node, package_layout):
+                break
```

Comment:
This "dirty" cleaning looks weird. Is this expected to have a dirty package? Why checking here? Are we checking the dirtyness everytime we access a package in the cache?

Reply:
This is not really new, mostly refactored.
Yes, the checks need to be here, it is where you first check if a given package binary exists in the cache. If that package was dirty from a previous build or whatever, it needs to be cleared. A different story is that now with revisions it is less likely because they are built in a different place in the cache, then moved to the final folder.